extensions/wikia/AdEngine/js/template/bfaaDesktop.js

```diff
@@ -72,20 +71,37 @@ define('ext.wikia.adEngine.template.bfaaDesktop', [
 			recoveryTweaker.tweakSlot(params.slotName, iframe);
 		}
 
-		if (params.videoUrl && params.videoTriggerElement) {
-			slotTweaker.onReady(params.slotName, function() {
-				var divs = doc.querySelectorAll('#' + params.slotName + ' > div'),
-					imageContainer = divs[divs.length - 1],
-					video = uapVideoAd.init(doc.getElementById(params.slotName), imageContainer, params.videoUrl);
-
-				params.videoTriggerElement.addEventListener('click', function () {
-					uapVideoAd.playAndToggle(video, imageContainer);
-				});
+		if (params.videoTriggerElement && params.videoAspectRatio) {
+			videoAdFactory.init().then(function () {
+				try {
+					var video = videoAdFactory.create(
+						adSlot.querySelector('div:last-of-type'),
+						function () {
+							return document.body.clientWidth;
+						},
+						function () {
+							return document.body.clientWidth / params.videoAspectRatio;
+						},
+						adSlot,
+						{
+							src: 'gpt',
+							slotName: params.slotName,
+							uap: params.uap,
+							passback: 'vuap'
+						}
+					);
+
+					params.videoTriggerElement.addEventListener('click', video.play.bind(video));
+
+				} catch (error) {
+					log(['Video can\'t be loaded correctly', error.message], log.levels.warning, logGroup);
```

Comment:
In most of the file, we're using a string as a log level: `'info'`, [feel free to use here string as well](https://github.com/Wikia/app/blob/dev/resources/wikia/modules/log.js#L136) (`'warning'`) or change all `'info'` occurrences to `log.levels.info`. Either way but consistent across the file, please :)